File: solve.rb
day = "day" + ARGV[0]
part = ARGV[1] || 1
file = ARGV[2] || "input"

input = File.open(day + "/" + file).readlines(&:chomp).map{|l| l.gsub("\n","")}.map(&:to_i)

require_relative(day + "/part" + part.to_s + ".rb")
puts solve(input)

File: solve.js
const fs = require("fs");
const relative = require("require-relative");
const watch = require("node-watch");
const args = process.argv.slice(2);
const day = "day" + ((args && parseInt(args[0])) || 1);
const file = "part" + ((args && parseInt(args[1])|| 1) + ".js")
const mod_path = ".\\" + day + "\\" + file;

watch(
  __dirname + "\\" + day,
  { recursive: true },
  (evt, name) => {
    delete require.cache[require.resolve(mod_path)];

    const process = (filename) => {
      let input;
      try {
        input = fs
          .readFileSync(__dirname + "\\" + day + "\\" + filename, "utf-8")
          .toString()
          .split("\n")
          .map((x) => x.replace("\r", ""));
      } catch (error) {
        return;
      }

      try {
        relative(mod_path).solve(input);
      } catch (error) {
        console.log(error);
      }
    };

    process("input_simple");
    process("input");
  }
);
